### ISSUE TITLE
Use transportSecurity-1.0 for appSecurity-4.0

### ISSUE THREAD

**Issue by jhanders34** [OpenLiberty/open-liberty#18855]
ssl-1.0 has been superseded by transportSecurity-1.0 for quite some time now.  appSecurity-1.0 through 3.0 all depend on ssl-1.0.  With Jakarta EE 9, should we switch to use transportSecurity-1.0 in appSecurity-4.0?  Additionally should we switch to not have transportSecurity-1.0 depend on ssl-1.0 any longer to remove some confusion?  (Not sure if we would consider this a breaking change).

Pros:  
1. Jakarta EE 9 features do not depend on a feature that has been superseded for years.  It only makes sense that newly created features should not depend on superseded features.
2. In the end transportSecurity-1.0 pulls in ssl-1.0 so it is somewhat confusing.

Cons:
1. Applications may need to be updated to use SSLContext vs SSLFactory to configure things.  As such it is a breaking change, but it is introduced with a new version of the appSecurity feature, so it is the right time to introduce the change.
2. You can no longer choose one or the other if you have appSecurity-4.0 feature enabled.  Previously customers could run with appSecurity and if they wanted the extra function of transportSecurity, they could just add the transportSecurity-1.0 feature to get the new behavior.
